application/F3DStarter.cxx

```diff
@@ -330,47 +332,171 @@ void F3DStarter::LoadFile(f3d::loader::LoadFileEnum load)
+    F3DAppOptions fileAppOptions = this->Internals->AppOptions;
+    this->Internals->Parser.UpdateOptions(filePath.string(), fileAppOptions,
+      this->Internals->FileOptions, this->Internals->UpdateWithCommandLineParsing);
+    this->Internals->UpdateWithCommandLineParsing = false; // this is done only once
+    this->Internals->Engine->setOptions(this->Internals->FileOptions);
+
+    this->Internals->LoadedFile = false;
+
+    // Check the size of the file before loading it
+    // Not considered in the context of GroupGeometries
+    static constexpr int BYTES_IN_MIB = 1048576;
+    if (fileAppOptions.MaxSize >= 0.0 &&
+      fs::file_size(filePath) > static_cast<std::uintmax_t>(fileAppOptions.MaxSize * BYTES_IN_MIB))
     {
-      this->Internals->Engine->getWindow().setWindowName(fileName + " - " + F3D::AppName);
+      f3d::log::info("No file loaded, file is bigger than max size");
+      filenameInfo = "No file loaded, file is bigger than max size";
+    }
+    else
+    {
+      try
+      {
+        if (loader.hasSceneReader(filePath.string()) && !fileAppOptions.GeometryOnly &&
+          !fileAppOptions.GroupGeometries)
+        {
+          loader.loadFullScene(filePath.string());
+          this->Internals->LoadedFile = true;
+        }
+        else if (loader.hasGeometryReader(filePath.string()))
+        {
+          // In GroupGeometries, just load all the files from the list
+          if (fileAppOptions.GroupGeometries)
+          {
+            int nGeom = 0;
+            for (size_t i = 0; i < size; i++)
+            {
+              auto geomPath = this->Internals->FilesList[i];
+              if (loader.hasGeometryReader(geomPath.string()))
+              {
+                // Reset for the first file, then add geometries without resetting
+                loader.loadGeometry(this->Internals->FilesList[i].string(), i == 0 ? true : false);
+                nGeom++;
+              }
+              else
+              {
+                f3d::log::warn(geomPath, " is not a geometry of a supported file format\n");
+              }
+            }
+            if (nGeom > 0)
+            {
+              filenameInfo = std::to_string(nGeom) + " Geometries loaded";
+            }
+            else
+            {
+              f3d::log::info("No geometry loaded");
+              filenameInfo = "No geometry loaded";
+            }
+          }
+          else
+          {
+            // Standard loadGeometry code
+            loader.loadGeometry(filePath.string(), true);
+          }
+          this->Internals->LoadedFile = true;
+        }
+        else
+        {
+          f3d::log::warn(filePath, " is not a file of a supported file format\n");
+          filenameInfo += " [UNSUPPORTED]";
+        }
+      }
+      catch (const f3d::loader::load_failure_exception& ex)
+      {
+        f3d::log::error("Could not load file:");
+        f3d::log::error(ex.what());
```

Comment:
can you combine these two lines? Maybe just `f3d::log::error("Could not load file: ", e.what());`